src/glang/lib/std/json.c3

```diff
@@ -449,22 +456,15 @@ function [IO] write : (io : IO mut&, node : JSON_Node) -> TypeIf<StringView, IsS
         // FIXME we should be using a buffer here.
         io:write(sv("{"));
 
-        mut iter : VectorIter<JSON_Node> = node.children:get_iter();
-        mut flag : bool = true;
+        mut iter : MapIter<JSON_Node, JSON_Node> = node.map:get_citer();
 
         // XXX we have to borrow the iterator, otherwise it is not updated
         // in place by the implicit call to get_next().
         for element in &mut iter {
-            io:write(element);
-
-            if flag {
-                io:write(sv(": "));
-            } else {
-                if iter:has_next() {
-                    io:write(sv(", "));
-                }
+            io:writef(sv("%: %"), element.key, element.value);
```

Comment:
~We should be quoting + escaping element keys right? That way we emit valid json as well as parsing it.~

Ohh, both the key and a value are a json_node, nvm then. This could be optimized to fix all keys as a string but thats a nit

Reply:
Done, and added a FIXME for the escaping bit